Command/MigrateCommand.php

```diff
@@ -120,57 +98,35 @@ protected function configure()
                 InputOption::VALUE_OPTIONAL,
                 'Query used to find elasticsearch records to import',
                 '{"sort":{"_uid":{"order":"asc"}}}'
+            )
+            ->addOption(
+                'dont-finalize',
+                null,
+                InputOption::VALUE_NONE,
+                'Don\'t finalize document'
             );
-        ;
-    }
-    
-    
-    private function getSubmitData(FormInterface $form)
-    {
-        return $this->dataService->getSubmitData($form);
-    }
-    
-    private function getEmptyRevision(ContentType $contentType)
-    {
-        return $this->dataService->getEmptyRevision($contentType, 'SYSTEM_MIGRATE');
     }
 
-    /**
-     * @param InputInterface $input
-     * @param OutputInterface $output
-     * @return int|void|null
-     * @throws MappingException
-     * @throws ORMException
-     * @throws OptimisticLockException
-     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        /** @var EntityManager $em */
-        $em = $this->doctrine->getManager();
-        //https://stackoverflow.com/questions/9699185/memory-leaks-symfony2-doctrine2-exceed-memory-limit
-        //https://stackoverflow.com/questions/13093689/is-there-a-way-to-free-the-memory-allotted-to-a-symfony-2-form-object
-        $em->getConnection()->getConfiguration()->setSQLLogger(null);
-        
-        
-        $signData = $input->getOption('sign-data');
-        
         $elasticsearchIndex = $input->getArgument('elasticsearchIndex');
         $contentTypeNameFrom = $input->getArgument('contentTypeNameFrom');
         $contentTypeNameTo = $input->getArgument('contentTypeNameTo');
+        $scrollSize = $input->getArgument('scrollSize');
+        $bulkSize = $input->getOption('bulkSize');
+        $scrollTimeout = $input->getArgument('scrollTimeout');
+        $rawImport = $input->getOption('raw');
+        $forceImport = $input->getOption('force');
+        $signData = $input->getOption('sign-data');
         $searchQuery = $input->getOption('searchQuery');
-        if (!$contentTypeNameTo) {
+        $finalize = !$input->getOption('dont-finalize');
+
+        if ($contentTypeNameTo === null) {
             $contentTypeNameTo = $contentTypeNameFrom;
         }
-        $scrollSize = $input->getArgument('scrollSize');
-        $scrollTimeout = $input->getArgument('scrollTimeout');
 
-        $this->bulker
-            ->setLogger(new ConsoleLogger($output))
-            ->setSize($input->getOption('bulkSize'));
-        
-        /** @var RevisionRepository $revisionRepository */
-        $revisionRepository = $em->getRepository('EMSCoreBundle:Revision');
-        /** @var ContentTypeRepository $contentTypeRepository */
+        /** @var EntityManager $em */
+        $em = $this->doctrine->getManager();
```

Comment:
part of initialization, every time the command runs we will need this.

Reply:
f2e7b00045703e33699bbc449ed23e8f5429907a

Reply:
I meant part of the consctructor (as $this->doctrine is not used for anything else) --> doctrine variable can be removed

Reply:
928fcdb08c0f8d687c4b09e9c2eb5deb7ae85a31